src/app/components/layouts/card/terra-card.component.spec.ts

```diff
@@ -1,3 +1,185 @@
+        cardComponent.inputPlaceholderIcon = null;
+        cardComponent.inputImagePath = null;
+    });
+
+    it('should create', () =>
+    {
+        expect(component).toBeTruthy();
+    });
+
+    it('should initialize its inputs', () =>
+    {
+        expect(cardComponent.inputIsSelected).toBe(false);
+    });
+
+    // footer
+    it('should div-element for footer be shown if content is given', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let footerElement:DebugElement = debugElement.query(By.css('div.card-footer'));
+        let ngContentElement:DebugElement = footerElement.query(By.css('p'));
+        expect(footerElement).toBeTruthy();
+        expect(cardComponent.viewChildFooter).toBeTruthy();
+        expect(cardComponent.viewChildFooter.nativeElement.children.length).toBeGreaterThan(0);
+        expect(ngContentElement.nativeElement.textContent.trim()).toEqual('card footer');
+        expect(footerElement.nativeElement.hidden).toBe(false);
+    });
+
+    it('should set class selected depending on #inputIsSelected', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let footerElement:DebugElement = debugElement.query(By.css('div.card-footer'));
+        let terraCardElement:DebugElement = debugElement.query(By.css('div.terra-card'));
+        expect(footerElement.classes['selected']).toBeFalsy();
+        expect(terraCardElement.classes['selected']).toBeFalsy();
+
+        cardComponent.inputIsSelected = true;
+        fixture.detectChanges();
+        expect(footerElement.classes['selected']).toBeTruthy();
+        expect(terraCardElement.classes['selected']).toBeTruthy();
+    });
+
+    // header
+    it('should show div-element for header if content is given', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let headerElement:DebugElement = debugElement.query(By.css('div.card-header'));
+        let ngContentElement:DebugElement = headerElement.query(By.css('p'));
+        expect(headerElement).toBeTruthy();
+        expect(cardComponent.viewChildHeader).toBeTruthy();
+        expect(cardComponent.viewChildHeader.nativeElement.children.length).toBeGreaterThan(0);
+        expect(ngContentElement.nativeElement.textContent.trim()).toEqual('card header');
+        expect(headerElement.nativeElement.hidden).toBe(false);
+    });
+
+    // block
+    it('should show div-element for content if content is given', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let contentElement:DebugElement = debugElement.query(By.css('div.card-block'));
+        let ngContentElement:DebugElement = contentElement.query(By.css('p'));
+        expect(contentElement).toBeTruthy();
+        expect(ngContentElement.nativeElement.textContent.trim()).toEqual('card content');
+    });
+
+    // image
+    it('should show image if #inputImagePath is set', () =>
+    {
+        let debugElement:DebugElement;
```

Comment:
Sofort initialisieren

Reply:
was initialisieren? das `debugElement` wird jetzt in `beforeEach`initialisiert